src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void resize() {
+        if (size == values.length) {
+            T[] buffer = (T[]) new Object[values.length + values.length / 2];
+            System.arraycopy(values, 0, buffer, 0, size);
+            values = buffer;
+        }
+    }
+
+    private void indexCheck(int index, int rangeMax) {
```

Comment:
You can use size instead of rangeMax

Reply:
Method indexCheck was optimized and now has only one parameter (index). Also method add was changed for that.